interfaces/apparmor/backend.go

```diff
@@ -529,18 +529,34 @@ func (b *Backend) SetupMany(snaps []*snap.Info, confinement func(snapName string
 	return errors
 }
 
-// Remove removes and unloads apparmor profiles of a given snap.
+// Removes all AppArmor profiles from disk but does not unload them from the
+// kernel - currently it is not possible to ensure that all services of a snap
+// are stopped and so it is not possible to safely unload the profile for a snap
+// from the kernel and so we only remove it from the cache on disk
+func RemoveAllSnapAppArmorProfiles() error {
+	dir := dirs.SnapAppArmorDir
+	globs := []string{"snap.*", "snap-update-ns.*", "snap-confine.*"}
+	cache := apparmor_sandbox.CacheDir
```

Comment:
I had a look at this but it would have meant rewriting all the callers of `RemoveCachedProfiles()` which I thought was a bit too much - but if you would prefer that I can definitely do it - let me know.